src/carplay.cs

```diff
@@ -0,0 +1,719 @@
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPListItem : NSSecureCoding {
+
+		[Field ("CPMaximumListItemImageSize")]
+		CGSize MaximumListItemImageSize { get; }
+
+		[Export ("initWithText:detailText:image:showsDisclosureIndicator:")]
+		IntPtr Constructor ([NullAllowed] string text, [NullAllowed] string detailText, [NullAllowed] UIImage image, bool showsDisclosureIndicator);
+
+		[Export ("initWithText:detailText:image:")]
+		IntPtr Constructor ([NullAllowed] string text, [NullAllowed] string detailText, [NullAllowed] UIImage image);
+
+		[Export ("initWithText:detailText:")]
+		IntPtr Constructor ([NullAllowed] string text, [NullAllowed] string detailText);
+
+		[NullAllowed, Export ("text")]
+		string Text { get; }
+
+		[NullAllowed, Export ("detailText")]
+		string DetailText { get; }
+
+		[NullAllowed, Export ("image", ArgumentSemantic.Strong)]
+		UIImage Image { get; }
+
+		[Export ("showsDisclosureIndicator")]
+		bool ShowsDisclosureIndicator { get; }
+
+		[NullAllowed, Export ("userInfo", ArgumentSemantic.Strong)]
+		NSObject UserInfo { get; set; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPListSection : NSSecureCoding {
+
+		[Export ("initWithItems:header:sectionIndexTitle:")]
+		IntPtr Constructor (CPListItem [] items, [NullAllowed] string header, [NullAllowed] string sectionIndexTitle);
+
+		[Export ("initWithItems:")]
+		IntPtr Constructor (CPListItem [] items);
+
+		[NullAllowed, Export ("header")]
+		string Header { get; }
+
+		[NullAllowed, Export ("sectionIndexTitle")]
+		string SectionIndexTitle { get; }
+
+		[Export ("items", ArgumentSemantic.Copy)]
+		CPListItem [] Items { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (CPTemplate))]
+	[DisableDefaultCtor]
+	interface CPListTemplate : CPBarButtonProviding {
+
+		[Export ("initWithSections:")]
+		IntPtr Constructor (CPListSection [] sections);
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		ICPListTemplateDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+
+		[Export ("sections", ArgumentSemantic.Copy)]
+		CPListSection [] Sections { get; }
+
+		[NullAllowed, Export ("title")]
+		string Title { get; set; }
+
+		[Export ("updateSections:")]
+		void UpdateSections (CPListSection [] sections);
+	}
+
+	interface ICPListTemplateDelegate { }
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))]
+	interface CPListTemplateDelegate {
+
+		[Abstract]
+		[Export ("listTemplate:didSelectListItem:completionHandler:")]
+		void DidSelectListItem (CPListTemplate listTemplate, CPListItem item, Action completionHandler);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	interface CPManeuver : NSCopying, NSSecureCoding {
```

Comment:
it's not clear to me if this is a user-created object ?
not sure the UI should be adding/modifying maneuvers

Reply:
hmm... there are setters everywhere so I guess it's mutable enough :)

Reply:
Yep, I think this one is fine as is